.github/workflows/demo/ec2-autoscaling-group.yaml

```diff
@@ -0,0 +1,159 @@
+      IamInstanceProfile: !Ref IamInstanceProfile
+      KeyName: !Ref KeyName
+      AssociatePublicIpAddress: true
+      SecurityGroups:
+        - !ImportValue
+          "Fn::Sub": "${VPCStack}-InstanceSecurityGroup"
+      EbsOptimized: true
+      BlockDeviceMappings:
+        - DeviceName: /dev/sda1
+          Ebs:
+            VolumeSize: 200
+
+      UserData:
+        Fn::Base64: !Sub |
+          #!/bin/bash -ex
+          mkdir -p /opt/aws/bin
+          wget https://s3.amazonaws.com/cloudformation-examples/aws-cfn-bootstrap-py3-latest.tar.gz
+          python3 -m easy_install --script-dir /opt/aws/bin aws-cfn-bootstrap-py3-latest.tar.gz
+
+          trap '/opt/aws/bin/cfn-signal --exit-code 1 --stack ${AWS::StackId} --resource AutoScalingGroup --region ${AWS::Region}' ERR
+
+          echo "Deploying Haystack demo, commit ${CommitShortSHA}"
+
+          echo 'APT::Periodic::Update-Package-Lists "0";
+          APT::Periodic::Unattended-Upgrade "0";' > /etc/apt/apt.conf.d/20auto-upgrades
+
+          apt update
+          apt install -y curl git ca-certificates curl gnupg lsb-release
+          curl -fsSL https://download.docker.com/linux/ubuntu/gpg | gpg --dearmor -o /usr/share/keyrings/docker-archive-keyring.gpg
+          echo \
+              "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/docker-archive-keyring.gpg] https://download.docker.com/linux/ubuntu \
+              $(lsb_release -cs) stable" | tee /etc/apt/sources.list.d/docker.list > /dev/null
+          apt update
+          apt install -y docker-ce docker-ce-cli containerd.io
+
+          # Install Docker compose
+          curl -L "https://github.com/docker/compose/releases/download/1.29.2/docker-compose-$(uname -s)-$(uname -m)" -o /usr/bin/docker-compose
+          chmod +x /usr/bin/docker-compose
+
+          # Install Nvidia container runtime
+          curl -s -L https://nvidia.github.io/nvidia-container-runtime/gpgkey | \
+            apt-key add -
+          distribution=$(. /etc/os-release;echo $ID$VERSION_ID)
+          curl -s -L https://nvidia.github.io/nvidia-container-runtime/$distribution/nvidia-container-runtime.list | \
+            tee /etc/apt/sources.list.d/nvidia-container-runtime.list
+          apt-get update
+          apt-get install -y nvidia-container-runtime
+
+          # Setup and start Docker
+          groupadd docker || true
+          usermod -aG docker $USER || true
+          newgrp docker || true
+          systemctl unmask docker
+          systemctl restart docker
+
+          # Exposes the GPUs to Docker
+          docker run --rm --gpus all ubuntu nvidia-smi
+
+          # Clone and start Haystack
+          git clone --branch deploy-demo https://github.com/askainet/haystack.git /opt/haystack
+          cd /opt/haystack
+          export COMPOSE_FILE=docker-compose-gpu.yml:.github/workflows/demo/docker-compose.demo.yml
+          docker-compose pull
+          docker-compose build
```

Comment:
Do we need `docker-compose build` ?

Reply:
Not really, as we are not trying to override the pulled images, and `docker-compose up` will build if required.